### DISCUSSION TITLE
Full featured version possible if Apple enables sideloading on iOS/iPadOS 17 in Europe?

### DISCUSSION THREAD

**Discussion by helmut72** [utmapp/UTM#5245] [General]
Is a full featured version possible if Apple enables sideloading on iOS/iPadOS 17 in Europe? Thank you.

**Comment by osy** [comment]
“Full featured” as in what you can get with TrollStore/jailbreak (Hypervisor, USB, etc), then no because they require private entitlements that App Store apps don’t get.

Even JIT supported UTM may not be possible depending on how apple implements it.

However, UTM SE would probably be able to be side loaded easily.

**Reply by helmut72** [reply]
Thanks for your answer.

**Comment by joeblack2k** [comment]
Now we have an answer, is the above statement still relevant/true?

thanks!

**Reply by osy** [reply]
Yes, Apple announced very strict rules about who gets JIT entitlements including that it must be a web browser where 90% of the screen is dedicated to rendering web content and it must pass some javascript compatibility tests. It is specifically written to block any non-web browser from using it.

**Reply by joeblack2k** [reply]
Thanks for the update.. i hate Apple

**Comment by kaystrobach** [comment]
how about other app stores? Afaik they can't block content there - correct? having UTM as webassembly would be genius, but will not be easily possible.

**Reply by osy** [reply]
The problem is that UTM requires JIT entitlements. Apple has indicated that even for third party marketplaces, they will only grant such entitlements for apps that meet a specific set of requirements the includes must pass JavaScript compliance tests and have most of the on screen content be web content controlled by the user.

**Reply by kaystrobach** [reply]
thank you for the explanation.

Maybe the point to ask the EU for help again. They have so powerfull devices and limit them to painting and other useless stuff. Imagine having a full blown linux on your journey with iOS - dreaming

Thanks a lot!

**Reply by helmut72** [reply]
It would be even enough for me if it's a native (aarch64) Linux Server. Using my iPad 99% of the time with Smart Keyboard.

**Comment by helmut72** [comment]
Apple silently released JIT APIs for iOS 17.4+ / iPadOS 18+:

https://developer.apple.com/documentation/apple-silicon/porting-just-in-time-compilers-to-apple-silicon

Is it now possible?

**Reply by osy** [reply]
Where does it say iOS?

**Comment by helmut72** [comment]
Unfortunately he was wrong. Got the info from here:
https://github.com/dotnet/runtime/issues/108423
